### ISSUE TITLE
Columns with default values are appended onto queries no matter what.

### ISSUE THREAD

**Issue by ImOliwer** [uptrace/bun#490]
Are columns that have default values yet no value assigned to said field supposed to be appended onto a query?
For example - take this table;

````go
type User struct {
  ID    string `bun:"id,notnull,type:uuid,default:gen_random_uuid()"`
  Email string `bun:",notnull"`
}
````

If I were to try and insert this accordingly, without including ID for said instance, for example 
`db.NewInsert().Model(&User{ Email: "email" }).Exec(context)` - it would cause the **22P02** (invalid text representation) **error** due to id not being specified and **emptied out** on **query appendices**.

Is there a way of handling this manually or ignoring fields on specifically **INSERT** operations?

**Comment by imraan-go** [comment]
Yes, You can use `bun:",nullzero"` tag to use DEFAULT in sql insert. The `default` tag is used for DDL only. Although I find it odd, @vmihailenco is there any way to improve it? There should be a better way to tell bun to ignore certain fields when inserting.

**Comment by ImOliwer** [comment]
> 

I see. That most definitely clears up my confusion, thanks. I probably should have looked a little more deeper into the documentation and have found `nullzero` in the `Defining models` section, silly mistake. I thought it was odd as well due to the fact that I assumed it automatically handled the field the way `nullzero` does, but with only `default` present in the tags since it more or less indicates that the field can be `NULL` or `DEFAULT` and to be ignored. But then, I guess there are cases where one would like to perhaps pass said `DEFAULT` value of a type purposely **WITHOUT** them being ignored. It makes sense, but if anyone ever comes up with a possibly better way, I think we're all for it. I will proceed with not closing this just in-case **vmihailenco** decides to fleek in on the matter.

**Comment by ImOliwer** [comment]
> Yes, You can use `bun:",nullzero"` tag to use DEFAULT in sql insert. The `default` tag is used for DDL only. Although I find it odd, @vmihailenco is there any way to improve it? There should be a better way to tell bun to ignore certain fields when inserting.

I see. That most definitely clears up my confusion, thanks. I probably should have looked a little more deeper into the documentation and have found `nullzero` in the `Defining models` section, silly mistake. I thought it was odd as well due to the fact that I assumed it automatically handled the field the way `nullzero` does, but with only `default` present in the tags since it more or less indicates that the field can be `NULL` or `DEFAULT` and to be ignored. But then, I guess there are cases where one would like to perhaps pass said `DEFAULT` value of a type purposely **WITHOUT** them being ignored. It makes sense, but if anyone ever comes up with a possibly better way, I think we're all for it. I will proceed with not closing this just in-case **vmihailenco** decides to fleek in on the matter.

**Comment by imraan-go** [comment]
> 
No scanonly doesnt do what you are looking for.  Take a look at #489

**Comment by vmihailenco** [comment]
>I will proceed with not closing this just in-case vmihailenco decides to fleek in on the matter.

I don't have better ideas how to handle this. Sometimes people want to marshal empty values as `NULL` and sometimes they don't. Explicit `nullzero` option allows to do both.

>No scanonly doesnt do what you are looking for.

I will slightly re-word docs for scanonly option.
